platforms/core-execution/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultCacheFactory.java

```diff
@@ -95,24 +95,18 @@ private PersistentCache doOpen(
         @Nullable Consumer<? super PersistentCache> initializer,
         CacheCleanupStrategy cacheCleanupStrategy
     ) {
-        File canonicalDir;
-        try {
-            canonicalDir = cacheDir.getCanonicalFile();
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-        DirCacheReference dirCacheReference = dirCaches.get(canonicalDir);
+        DirCacheReference dirCacheReference = dirCaches.get(cacheDir);
```

Comment:
❓ Woudn't that make it a bit more fragile depending how File path is constructed?
Example:
```
new File("./cache").equals(new File("cache")); // returns false
new File("./cache").getCanonicalFile().equals(new File("cache").getCanonicalFile()); // returns true
```

So `dirCaches.get(canonicalDir)` could return different results.

Reply:
Not sure if that is problem in practice though

Reply:
The canincalization was added in 2009, I don't think it matters. User code does not create these caches, so we control all creation. We've been canonicalizing the directory build cache's location already, that's why it didn't cause any trouble (Maven didn't, that's how we found the discrepancy).

In theory, we could canonicalize everywhere where we resolve a location, but that'd be defensive coding.